processing/src/test/java/org/apache/druid/segment/join/PostJoinCursorTest.java

```diff
@@ -0,0 +1,249 @@
+import org.joda.time.DateTime;
+import org.joda.time.Interval;
+import org.junit.Test;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.List;
+
+import static java.lang.Thread.sleep;
+import static org.junit.Assert.assertTrue;
+
+public class PostJoinCursorTest extends BaseHashJoinSegmentStorageAdapterTest
+{
+
+  public QueryableIndexSegment factSegment;
+
+  @Override
+  public void setUp() throws IOException
+  {
+
+    factSegment = new TestInfiniteQueryableIndexSegment(
+        JoinTestHelper.createFactIndexBuilder(temporaryFolder.newFolder()).buildMMappedIndex(),
+        SegmentId.dummy("facts")
+    );
+    countryIsoCodeToNameLookup = JoinTestHelper.createCountryIsoCodeToNameLookup();
+    countryNumberToNameLookup = JoinTestHelper.createCountryNumberToNameLookup();
+    countriesTable = JoinTestHelper.createCountriesIndexedTable();
+    regionsTable = JoinTestHelper.createRegionsIndexedTable();
+  }
+
+  /**
+   * Simulates infinite segment by using a base cursor with advance() and advanceInterruptibly()
+   * reduced to a no-op.
+   */
+  private static class TestInfiniteQueryableIndexSegment extends QueryableIndexSegment
+  {
+
+    private final StorageAdapter testStorageAdaptor;
+
+    public TestInfiniteQueryableIndexSegment(QueryableIndex index, SegmentId segmentId)
+    {
+      super(index, segmentId);
+      testStorageAdaptor = new InfiniteQueryableIndexStorageAdapter(index);
+    }
+
+    @Override
+    public StorageAdapter asStorageAdapter()
+    {
+      return testStorageAdaptor;
+    }
+  }
+
+  private static class InfiniteQueryableIndexStorageAdapter extends QueryableIndexStorageAdapter
+  {
+    public InfiniteQueryableIndexStorageAdapter(QueryableIndex index)
+    {
+      super(index);
+    }
+
+    @Override
+    public Sequence<Cursor> makeCursors(
+        @Nullable Filter filter,
+        Interval interval,
+        VirtualColumns virtualColumns,
+        Granularity gran,
+        boolean descending,
+        @Nullable QueryMetrics<?> queryMetrics
+    )
+    {
+      return super.makeCursors(filter, interval, virtualColumns, gran, descending, queryMetrics).map(
+          CursorNoAdvance::new);
+    }
+
+    private static class CursorNoAdvance implements Cursor
+    {
+      Cursor cursor;
+
+      public CursorNoAdvance(Cursor cursor)
+      {
+        this.cursor = cursor;
+      }
+
+      @Override
+      public ColumnSelectorFactory getColumnSelectorFactory()
+      {
+        return cursor.getColumnSelectorFactory();
+      }
+
+      @Override
+      public DateTime getTime()
+      {
+        return cursor.getTime();
+      }
+
+      @Override
+      public void advance()
+      {
+        // Do nothing to simulate infinite rows
+      }
+
+      @Override
+      public void advanceUninterruptibly()
+      {
+        // Do nothing to simulate infinite rows
+      }
+
+      @Override
+      public boolean isDone()
+      {
+        return cursor.isDone();
```

Comment:
shouldn't this return false? May not matter but given its infinite cursor.

Reply:
Updated